nrf53: fix RTC pretick power usage for events on RTC0

ankuns opened the pull request:
Fixes #63789

For RTC0 events the RTC1 pretick event was not cleared what caused the WDT to be not stopped. This resulted in increased power usage.

ankuns committed:
nrf53: fix RTC pretick power usage for events on RTC0

For RTC0 events the RTC1 pretick event was not cleared what caused the
WDT to be not stopped. This resulted in increased power usage.

Signed-off-by: Andrzej Kuroś <andrzej.kuros@nordicsemi.no>

anangl commented on soc/arm/nordic_nrf/nrf53/soc.c:
```diff
@@ -415,6 +415,10 @@ static void rtc_pretick_rtc_isr_hook(void)
 {
 	rtc_pretick_rtc_isr_hook();
+
+	if (IS_ENABLED(CONFIG_SOC_NRF53_RTC_PRETICK)) {
```
This `if` is redundant as the whole thing is inside https://github.com/zephyrproject-rtos/zephyr/blob/bf4652e842c56737b66fe4d50b4dcef41010a3bb/soc/arm/nordic_nrf/nrf53/soc.c#L377

And why not do this event clearing in `rtc_pretick_rtc_isr_hook` (it is common for RTC0 and RTC1)?

ankuns replied:
fixed

ankuns committed:
nrf53: fix RTC pretick for RTC rescheduling by other interrupts

It might happen that while some interrupt handler other than for RTC0
or RTC1 (e.g. for RADIO) is executed, the scheduled pretick CC triggers.
This starts pretick pulses due to the loop through IPC. The change
in pretick schedule did not stop the pretick pulses going through IPC
loop, what caused heavy increase in power consumption.

This commit fixes this behavior.
Added also clarifications for Kconfig option `SOC_NRF53_RTC_PRETICK`.

Signed-off-by: Andrzej Kuroś <andrzej.kuros@nordicsemi.no>

ankuns commented:
Pushed additional commit, marking DNM for a while

ankuns committed:
nrf5340: pretick decoupled from workaround anomaly 160

Coupling in code between workarounds for anomaly 160 and anomaly 165
(pretick) is decreased.

Signed-off-by: Andrzej Kuroś <andrzej.kuros@nordicsemi.no>

jhedberg commented:
@ankuns @anangl there doesn't seem to be any "Fixes #12345" entry. Please add one (required at this stage of the release process if you want this for 3.5)

ankuns commented:
@jhedberg edited description, added "Fixes #63789"
